### PR TITLE
[BACK-3115] Fix Dexcom task data issues

### PR DESCRIPTION
**Pull request by darinkrauss**
- Fix bugs with Dexcom task not updating various times
- Fix bug with Dexcom task creating erroneous cgmSettings
- Fix bug with Dexcom task data source and provider sessions
- Fix Dexcom task connection and disconnection issues
- Add tool to analyze Dexcom task issues
- Minor Makefile updates
- Add and update tests
- https://tidepool.atlassian.net/browse/BACK-3113
- https://tidepool.atlassian.net/browse/BACK-3114
- https://tidepool.atlassian.net/browse/BACK-3116
- https://tidepool.atlassian.net/browse/BACK-3117
- https://tidepool.atlassian.net/browse/BACK-3118

### CODE REVIEW

File: tools/dexcom_analyze/dexcom_analyze.go [hunk 1]
```diff
@@ -0,0 +1,1543 @@
...
+	case Issue_DataSource_UserID_Missing:
+	case Issue_DataSource_With_DataSetIDs_DataSetIDs_Length_Invalid:
+	case Issue_DataSource_With_DataSetIDs_EarliestDataTime_Missing, Issue_DataSource_With_DataSetIDs_LastImportTime_Missing, Issue_DataSource_With_DataSetIDs_LatestDataTime_Missing:
+		t.outputResolutionHeader("FIXED with BACK-3113. Will keep occurring until deployed. Will need to manually fix.")
+		// t.outputMongoReadOperationsHeader()
+
+		// var earliestDataTimeIDs IDs
+		// if otherMarshable, ok := issueMarshalableMap[Issue_DataSource_With_DataSetIDs_EarliestDataTime_Missing]; ok {
+		// 	earliestDataTimeIDs = otherMarshable.DataSources().IDs()
+		// }
```
Comment (jh-bate) [inline tools/dexcom_analyze/dexcom_analyze.go:1190]: all the commented out sections?

**Review by jh-bate — COMMENTED** [review]
One main question - is usage of the dexcom analyze tool self evident enough that it don't require docs beyond the the the cli help would give?

File: tools/dexcom_analyze/dexcom_analyze.go (hunk 1 above)
Comment (darinkrauss) [reply tools/dexcom_analyze/dexcom_analyze.go:1214]: Those are intended to provide more details, if desired. I added a NOTE comment above each to indicate this.

**Comment by darinkrauss** [comment]
I added documentation to the dexcom_analyze tool.

**Review by jh-bate — APPROVED** [review]
LGTM

**Review by toddkazakov — APPROVED** [review]

**Comment by darinkrauss** [comment]
Closing. Included in grouped PR https://github.com/tidepool-org/platform/pull/764.

**Pull request closed without merging**